public/src/bowling.js

```diff
@@ -0,0 +1,102 @@
+class Bowling {
+  constructor() {
+    this.frameScore = []
+    this.frameStatus = ""
+    this.score = []
+    this.prevFrame = ""
+    this.frameCounter = 0
+  }
+
+  frame(score1, score2 = 0) {
+    this.frameScore = []
+    if ((Number(score1) === 10) && (Number(score2) > 0)) {
+      throw new Error('You may not bowl again in this frame');
+    }
+    if ((this._bowl1(score1) != 'strike') && (this._bowl2(score2) != 'spare')) {
+      this.score.push(score1, score2)
+      // return this._bowl2(score2)
+    }
+    else if (this._bowl2(score2) == 'spare') {
+      this._bowl2(score2)
+    }
+    else {
+      this._bowl1(score1)
+    };
+    this.frameCounter++
+    return this.bonus()
+  };
+
+  bonus() {
+    // prevFrame = spare
+    if ((this.frameStatus != 'strike') && (this.prevFrame == 'spare')) {
+      this.score.splice(-2, 0, (10 + Number(this.frameScore[0])))
+      this.score
+    };
+    if ((this.frameStatus == 'strike') && (this.prevFrame == 'spare')) {
+      this.score.push(20)
+      this.score
+    };
+    // prevframe = strike
+    if ((this.frameStatus == 'spare') && (this.prevFrame == 'strike')) {
+      this.score.push(20)
+    };
+    if ((this.frameStatus == 'strike') && (this.prevFrame == 'strike')) {
+      return this.prevFrame = '2xstrike'
+    };
+    //prevFrame = 2xstrike
+    if ((this.frameStatus == 'strike') && (this.prevFrame == '2xstrike')) {
+      this.score.push(30)
+    };
+    if ((this.frameStatus == 'spare') && (this.prevFrame == '2xstrike')) {
+      this.score.push(20 + Number(this.frameScore[0]))
+      this.score.push(20)
+    }
+    if ((this.frameStatus == 'openFrame') && (this.prevFrame == '2xstrike')) {
+      this.score.splice(-3, 0, (20 + Number(this.frameScore[0])))
+      this.score.splice(-2, 0, (10 + Number(this.frameScore[0]) + Number(this.frameScore[1])))
+    };
+    this.prevFrame = this.frameStatus
+  };
```

Comment:
I think branching here would make the logic easier to parse as there are many conditionals which have some common conditions